creator-node/test/tracks.js

```diff
@@ -382,3 +386,98 @@ describe('test Tracks', function () {
       .expect(200)
   })
 })
+
+describe('test /track_content with actual ipfsClient', function () {
+  let app, server, session, ipfs, libsMock
+
+  beforeEach(async () => {
+    ipfs = require('../src/ipfsClient').ipfs
+    libsMock = getLibsMock()
+
+    const appInfo = await getApp(ipfs, libsMock, blacklistManager)
+    await blacklistManager.blacklist(ipfs)
+
+    app = appInfo.app
+    server = appInfo.server
+    session = await createStarterCNodeUser()
+  })
+
+  afterEach(async () => {
+    sinon.restore()
+    await server.close()
+  })
+
+  it('sends server error response if segmenting fails', async function () {
+    const file = fs.readFileSync(testAudioFilePath)
+    sinon.stub(TranscodingQueue, 'segment').rejects(new Error('failed to segment'))
+
+    await request(app)
+      .post('/track_content')
+      .attach('file', file, { filename: 'fname.mp3' })
+      .set('Content-Type', 'multipart/form-data')
+      .set('X-Session-ID', session)
+      .expect(500)
+  })
+
+  it('sends server error response if transcoding fails', async function () {
+    const file = fs.readFileSync(testAudioFilePath)
+    sinon.stub(TranscodingQueue, 'transcode320').rejects(new Error('failed to transcode'))
+
+    await request(app)
+      .post('/track_content')
+      .attach('file', file, { filename: 'fname.mp3' })
+      .set('Content-Type', 'multipart/form-data')
+      .set('X-Session-ID', session)
+      .expect(500)
+  })
+
+  it('should upload 32 segments and 1 320kbps copy to storagePath', async function () {
+    const file = fs.readFileSync(testAudioFilePath)
+    libsMock.User.getUsers.exactly(4)
+
+    const resp = await request(app)
+      .post('/track_content')
+      .attach('file', file, { filename: 'fname.mp3' })
+      .set('Content-Type', 'multipart/form-data')
+      .set('X-Session-ID', session)
+      .expect(200)
+
+    let storagePath = config.get('storagePath')
+    storagePath = storagePath.slice(0, 1) === '/' ? '.' + storagePath : storagePath
+
+    // check if track UUID dir exists
+    const originalTrackUUID = resp.body.source_file.slice(0, -4) // trim off .mp3
```

Comment:
a more generic way to do this would be split the string on the last period and remove the extension. like `filename.split('.').slice(0, -1).join('.')`